src/main/jbake/content/blog/2017/01/13/events-aggregation-extension.adoc

```diff
@@ -0,0 +1,189 @@
+= Extending Complex Event Processing in Hawkular Alerting
+Lucas Ponce
+2017-01-13
+:jbake-type: post
+:jbake-status: published
+:jbake-tags: blog, alerts, complex, events, processing, stream, polling
+
+== Stream Processing versus Polling Processing
+
+Hawkular Alerting uses different techniques to detect behaviours by defining rules.
+
+The main implementation is based in a Stream Processing design.
+Hawkular Alerting analyzes streams of data and events ordered in a timeline searching for matches against conditions.
+This technique is also known as Event Processing (or Complex Event Processing) specially when the processing of incoming data might need multiple conditions.
```

Comment:
One can argue that event processing can be prefixed by word "Complex" only if the system allows to emit the complex events (high lvl) derived from the simple events (low lvl). While at the same time the emitted complex events can be input to other event processors. "Complex" is not related to how complicated it is or how the conditions can be rich.., imho.

But I guess there is no mathematical definition out there, so we should be good. I am just saying that I have different understanding what CEP actually means.